python/taichi/examples/algorithm/poisson_disk_sampling.py

```diff
@@ -0,0 +1,163 @@
+            ind = grid[i, j]
+            if ind != -1:
+                q = samples[ind]
+                d = (q - p).norm()
+                if d < dmin:
+                    dmin = d
+                    nearest = q
+    return dmin, nearest
+
+
+@ti.kernel
+def poisson_disk_sample(num_samples: int) -> int:
+    while head[None] < tail[None] and head[None] < ti.min(
+            num_samples, desired_samples):
+        source_x = samples[head[None]]
+        head[None] += 1
+
+        for _ in range(100):
+            theta = ti.random() * 2 * tm.pi
+            offset = tm.vec2(tm.cos(theta),
+                             tm.sin(theta)) * (1 + ti.random()) * radius
+            new_x = source_x + offset
+            new_index = coord_to_index(new_x)
+
+            if 0 <= new_x[0] < 1 and 0 <= new_x[1] < 1:
+                collision = (find_nearest_point(new_x)[0] < radius - 1e-6)
+                if not collision and tail[None] < desired_samples:
+                    samples[tail[None]] = new_x
+                    grid[new_index] = tail[None]
+                    tail[None] += 1
+    return tail[None]
+
+
+@ti.func
+def hash21(p):
+    return tm.fract(tm.sin(tm.dot(p, tm.vec2(127.619, 157.583))) * 43758.5453)
+
+
+@ti.func
+def sample_dist(uv):
+    uv = uv * iResolution
+    x, y = tm.clamp(0, iResolution - 1, uv).cast(int)
+    return dfield[x, y]
+
+
+@ti.kernel
+def compute_distance_field():
+    for i, j in dfield:
+        uv = tm.vec2(i, j) / iResolution
+        d, p = find_nearest_point(uv)
+        d = (uv - p).norm() - radius / 2.
+        dfield[i, j] = tm.vec4(d, p.x, p.y, radius / 2.)
+
+
+@ti.kernel
+def render():
+    for i, j in img:
+        uv = tm.vec2(i, j) / iResolution.y
+        st = tm.fract(uv * grid_n) - 0.5
+        dg = 0.5 - abs(st)
+        d1 = ti.min(dg.x, dg.y)
+        d1 = tm.smoothstep(0.05, 0.0, d1)
+        col = (1 - tm.vec3(d1)) * 0.7
+        sf = 2 / iResolution.y
+        buf = sample_dist(uv)
+        bufSh = sample_dist(uv + tm.vec2(0.005, 0.015))
+        cCol = tm.vec3(hash21(buf.yz + 0.3), hash21(buf.yz),
+                       hash21(buf.yz + 0.09))
+        pat = (abs(tm.fract(-buf.x * 150) - 0.5) * 2) / 300
+        col = tm.mix(col, tm.vec3(0),
+                     (1 - tm.smoothstep(0, 3 * sf, pat)) * 0.25)
+        ew, ew2 = 0.005, 0.008
+        cCol2 = tm.mix(cCol, tm.vec3(1), 0.9)
+        col = tm.mix(col, tm.vec3(0),
+                     (1 - tm.smoothstep(0, sf * 2, bufSh.x)) * 0.4)
+        col = tm.mix(col, tm.vec3(0), 1 - tm.smoothstep(sf, 0, -buf.x))
+        col = tm.mix(col, cCol2, 1 - tm.smoothstep(sf, 0, -buf.x - ew))
+        col = tm.mix(col, tm.vec3(0),
+                     1 - tm.smoothstep(sf, 0, -buf.x - ew2 - ew))
+        col = tm.mix(col, cCol,
+                     1 - tm.smoothstep(sf, 0., -buf.x - ew2 - ew * 2))
+        col = tm.sqrt(ti.max(col, 0))
+        img[i, j] = col
+
+
+refresh_scene()
```

Comment:
nit: please add a main function for the script and doing things in the global scope in the script.